src/Microsoft.DotNet.Interactive.Http/AspNetExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Reflection;
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DotNet.Interactive.Http
+{
+    public static class AspNetExtensions
+    {
+        public static IServiceCollection AddKernel<T>(this IServiceCollection services, T kernel) where T : Kernel
+        {
+            services.AddSingleton(kernel);
+            services.AddSingleton<Kernel>(kernel);
+
+            return services;
+        }
+
+        public static IWebHostBuilder UseDotNetInteractive(this IWebHostBuilder builder, bool enableHttpApi,
+            HttpPort httpPort, HttpProbingSettings httpProbingSettings, IServiceCollection services)
+        {
+            var httpStartupOptions = new HttpOptions(enableHttpApi, httpPort);
+
+            builder.ConfigureServices(c =>
+                {
+                    c.AddSingleton(httpStartupOptions);
+                })
+                .ConfigureServices(c =>
+                {
+                    if (enableHttpApi && httpProbingSettings != null)
+                    {
+                        c.AddSingleton(httpProbingSettings);
+                    }
+
+                    c.AddSingleton(httpStartupOptions);
+
+                    if (services != null)
+                    {
+                        foreach (var serviceDescriptor in services)
+                        {
+                            c.Add(serviceDescriptor);
+                        }
+                    }
+                });
+            return builder;
+        }
+        public static IServiceCollection AddDotnetInteractive(this IServiceCollection services)
```

Comment:
Maybe the naming should also include the fact that this is adding HTTP-specific functionality?